win_dsc - Add argument validation and other fixes (#53093)
ansible/ansible: docs/docsite/rst/user_guide/windows_dsc.rst

        Ensure: Present
        Key: HKEY_LOCAL_MACHINE\SOFTWARE\ExampleKey
        ValueName: TestValue

>         ValueData: TestDatai

Reviewer: ```suggestion
        ValueData: TestData
```

> In Ansible 2.8, the ``win_dsc`` module will now automatically validate the

Reviewer: ```suggestion
Starting in Ansible 2.8, the ``win_dsc`` module automatically validates the
```

input options from Ansible with the DSC definition. This means Ansible will
fail if the option name is incorrect, a mandatory option is not set, or the

> value is not a valid choice. When running Ansible with a verbosity level or 3

Reviewer: ```suggestion
value is not a valid choice. When running Ansible with a verbosity level of 3
```

            "Time taken for configuration job to complete is 0.402 seconds"
        ],
        "verbose_test": [
            "Perform operation 'Invoke CimMethod' with following parameters, ''methodName' = ResourceTest,'className' = MSFT_DSCLocalConfigurationManager,'namespaceName' = root/Microsoft/Windows/DesiredStateConfiguration'.",
            "An LCM method call arrived from computer SERVER2016 with user sid S-1-5-21-3088887838-4058132883-1884671576-1105.",
            "[SERVER2016]: LCM:  [ Start  Test     ]  [[Registry]DirectResourceAccess]",
            "[SERVER2016]:                            [[Registry]DirectResourceAccess] Registry key 'HKLM:\\SOFTWARE\\ExampleKey' does not exist",
            "[SERVER2016]: LCM:  [ End    Test     ]  [[Registry]DirectResourceAccess] False in 0.2510 seconds.",
            "[SERVER2016]: LCM:  [ End    Set      ]    in  0.3310 seconds.",
            "Operation 'Invoke CimMethod' complete.",
            "Time taken for configuration job to complete is 0.475 seconds"
        ]
    }

The ``invocation.module_args`` key shows the actual values that were set as
well as other possible values that were not set. Unfortunately this will not

> show the default value for a DSC property just what was set from the Ansible

Reviewer: ```suggestion
show the default value for a DSC property, only what was set from the Ansible
```

task. Any ``*_password`` option will be masked in the output for security

> reasons, if there are any other sensitive module options set ``no_log: True``

Reviewer: ```suggestion
reasons, if there are any other sensitive module options, set ``no_log: True``
```

> on the task to stop it from being logged.

Reviewer: ```suggestion
on the task to stop all task output from being logged.
```
